shell/platform/android/io/flutter/embedding/engine/systemchannels/LifecycleChannel.java

```diff
@@ -15,23 +16,69 @@
  */
 public class LifecycleChannel {
 
+  public enum AppLifecycleState { INACTIVE, RESUMED, PAUSED }
+
   @NonNull
   public final BasicMessageChannel<String> channel;
 
+  /**
+   * Cache the current state so that any queries for it may access it.
+   */
+  private AppLifecycleState state = AppLifecycleState.INACTIVE;
```

Comment:
Please keep all system channels stateless.  The purpose of these channels is to codify the comm API, not add behavior.  Please follow examples of `TextInputPlugin`, `AndroidKeyProcessor`, and other areas where state surrounds a channel.